driver/compiler_hooks.ml

```diff
@@ -125,6 +128,9 @@ let register : type a. a pass -> (a -> unit) -> unit =
   | Cfg -> hooks.cfg <- f :: hooks.cfg
   | Cmm -> hooks.cmm <- f :: hooks.cmm
   | Inlining_tree -> hooks.inlining_tree <- f :: hooks.inlining_tree
+  | Check_allocations ->
+    Flambda_backend_flags.checkmach_details_cutoff := -1;
```

Comment:
I am a bit uneasy with the modification
of flag values here; could we rather use
`Misc.protect_refs` in `execute` below?